### PR TITLE
python3-mypy: add missing dependency on python3-typed-ast

### PR DESCRIPTION
**Pull request by whoizit**
```
$ mypy -h
Traceback (most recent call last):
  File "/usr/bin/mypy", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/user/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3126, in <module>
    @_call_aside
  File "/home/user/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3110, in _call_aside
    f(*args, **kwargs)
  File "/home/user/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3139, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/user/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 581, in _build_master
    ws.require(__requires__)
  File "/home/user/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 898, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/user/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 784, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'typed-ast<1.4.0,>=1.3.1' distribution was not found and is required by mypy
```

### CODE REVIEW

File: srcpkgs/python3-mypy/template [hunk 1]
```diff
@@ -7,7 +7,7 @@ wrksrc="mypy-${version}"
 build_style=python3-module
 pycompile_module="mypy"
 hostmakedepends="python3-setuptools"
-depends="python3-mypy_extensions"
+depends="python3-mypy_extensions python3-typed-ast"
```
Comment (maxice8) [inline srcpkgs/python3-mypy/template]: need to bump revision as well

**Comment by maxice8** [comment]
- poor commit name, try something along the lines of `python3-mypy: add missing dependency on python3-typed-ast`

File: srcpkgs/python3-typed-ast/template [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'python3-typed-ast'
+pkgname=python3-typed-ast
+version=1.3.1
+revision=1
+archs=noarch
+wrksrc="typed-ast-${version}"
+build_style=python3-module
+hostmakedepends="python3-devel python3-setuptools"
+depends="python3"
```
Comment (maxice8) [inline srcpkgs/python3-typed-ast/template]: no need to depend on python3, you are compiling `CPython` modules so they call into the python library will be detected and python will be added to the dependencies automatically.

File: srcpkgs/python3-typed-ast/template [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'python3-typed-ast'
+pkgname=python3-typed-ast
+version=1.3.1
+revision=1
+archs=noarch
```
Comment (maxice8) [inline srcpkgs/python3-typed-ast/template]: remove this, you are compiling `CPython` modules which is machine code.

File: srcpkgs/python3-typed-ast/template [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+wrksrc="typed-ast-${version}"
+build_style=python3-module
```
Comment (maxice8) [inline srcpkgs/python3-typed-ast/template]: missing pycompile_module="typed_ast"

File: srcpkgs/python3-typed-ast/template (hunk 2 above)
Comment (whoizit) [reply srcpkgs/python3-typed-ast/template]: in `python3-mypy_extensions` you did exactly the same

**Comment by maxice8** [comment]
On Wed, Mar 20, 2019 at 12:12:59AM -0700, whoami wrote:
> whoizit commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,19 @@
> +# Template file for 'python3-typed-ast'
> +pkgname=python3-typed-ast
> +version=1.3.1
> +revision=1
> +archs=noarch
> +wrksrc="typed-ast-${version}"
> +build_style=python3-module
> +hostmakedepends="python3-devel python3-setuptools"
> +depends="python3"
> 
> in `python3-mypy_extensions` you did exactly the same
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10066#discussion_r267207503

Because python3-mypy_extensions is not compiling CPython modules, it is a PURE python package without any dependency on any other library, so you need to add python to the dependencies.

Please read up on the Manual: https://github.com/void-linux/void-packages/blob/master/Manual.md#runtime-dependencies

**Comment by whoizit** [comment]
@maxice8 
```
[cc-wrapper] ignoring -L/usr/lib
/usr/lib/gcc/aarch64-linux-gnu/8.3.0/../../../../aarch64-linux-gnu/bin/ld: cannot find -lpython3.6m
collect2: error: ld returned 1 exit status
error: command 'aarch64-linux-gnu-gcc' failed with exit status 1
```

File: srcpkgs/python3-typed-ast/template [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'python3-typed-ast'
+pkgname=python3-typed-ast
+version=1.3.1
+revision=1
+wrksrc="typed-ast-${version}"
+build_style=python3-module
+pycompile_module="typed_ast"
+hostmakedepends="python3-devel python3-setuptools"
```
Comment (maxice8) [inline srcpkgs/python3-typed-ast/template]: python3-devel should be in makedepends as you are linking against the python library.

Packages that have libraries you are linking against go into makedepends.

Please read up on the Manual: https://github.com/void-linux/void-packages/blob/master/Manual.md#explain_depends

**Comment by whoizit** [comment]
@maxice8

**Comment by maxice8** [comment]
there are cross problems.

**Comment by maxice8** [comment]
Please mark it as nocross= for now

**Comment by whoizit** [comment]
@maxice8 done

**Comment by maxice8** [comment]
sorry for the long wait

**Pull request merged**